File: bahasaKorea/app/src/main/java/com/im/daeseong/bahasakorea/Database/KataManager.java
package com.im.daeseong.bahasakorea.Database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteException;
import android.util.Log;

import java.util.ArrayList;

public class KataManager {

    private static final String TAG = KataManager.class.getSimpleName();

    private DatabaseHelper databaseHelper = null;

    private static final String TABLE_KATAS = "Katas";

    private static KataManager instance = null;
    public static KataManager getInstance(Context context) {
        if (instance == null) {
            instance = new KataManager(context);
        }
        return instance;
    }

    private KataManager(Context context) {
        databaseHelper = DatabaseHelper.getInstance(context);
    }

    public void setKata(KataItems items){

        if(databaseHelper == null){
            return;
        }

        SQLiteDatabase database = databaseHelper.getWritableDatabase();

        try {

            ContentValues values = new ContentValues();
            values.put("rIndex", items.getrIndex());
            values.put("KataKor", items.getKataKor());
            values.put("KataIndo", items.getKataIndo());
            values.put("KataEng", items.getKataEng());
            database.insert("Katas", null, values);

        } catch (SQLiteException e){
        } catch (Exception e){
        } finally {
            if(database != null){
                database.close();
            }
        }
    }

    public int getTotalcount(){

        int nTotalcount = 0;

        if(databaseHelper == null){
            return nTotalcount;
        }

        SQLiteDatabase database = databaseHelper.getReadableDatabase();
        if(database == null){
            return nTotalcount;
        }

        String query = "select count(*) from Katas";
        Cursor cursor = database.rawQuery(query, null);

        try{

            if(cursor.moveToFirst()) {
                cursor.moveToFirst();
                nTotalcount =cursor.getInt(0);
            }

        } catch (SQLiteException e){
        } catch (Exception e){
        } finally {

            if(cursor != null){
                cursor.close();
                cursor = null;
            }

            if(database != null){
                database.close();
            }
        }
        return nTotalcount;
    }

    public KataItems getKata(String sSearch){

        KataItems items = null;

        if(databaseHelper == null){
            return items;
        }

        SQLiteDatabase database = databaseHelper.getReadableDatabase();
        if(database == null){
            return items;
        }

        String query = "select * from Katas where KataKor " + " =  \"" + sSearch + "\"";
        Cursor cursor = database.rawQuery(query, null);

        try{

            if(cursor.moveToFirst()) {
                cursor.moveToFirst();
                int rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"));
                String KataKor = cursor.getString(cursor.getColumnIndex("KataKor"));
                String KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"));
                String KataEng = cursor.getString(cursor.getColumnIndex("KataEng"));
                items = new KataItems(rIndex, KataKor, KataIndo, KataEng);
            }

        } catch (SQLiteException e){
        } catch (Exception e){
        } finally {

            if(cursor != null){
                cursor.close();
                cursor = null;
            }

            if(database != null){
                database.close();
            }
        }
        return items;
    }

    public KataItems getKata(int nIndex){

        KataItems items = null;

        if(databaseHelper == null){
            return items;
        }

        SQLiteDatabase database = databaseHelper.getReadableDatabase();
        if(database == null){
            return items;
        }

        String query = "select * from Katas where rIndex ="  + nIndex;
        Cursor cursor = database.rawQuery(query, null);

        try{

            if(cursor.moveToFirst()) {
                cursor.moveToFirst();
                int rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"));
                String KataKor = cursor.getString(cursor.getColumnIndex("KataKor"));
                String KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"));
                String KataEng = cursor.getString(cursor.getColumnIndex("KataEng"));
                items = new KataItems(rIndex, KataKor, KataIndo, KataEng);
            }

        } catch (SQLiteException e){
        } catch (Exception e){
        } finally {

            if(cursor != null){
                cursor.close();
                cursor = null;
            }

            if(database != null){
                database.close();
            }
        }
        return items;
    }

    public ArrayList<KataItems> getKata(){

        if(databaseHelper == null){
            return null;
        }

        SQLiteDatabase database = databaseHelper.getReadableDatabase();
        if(database == null){
            return null;
        }

        ArrayList<KataItems> list = new ArrayList<>();

        String query = "select * from Katas ";
        Cursor cursor = database.rawQuery(query, null);

        try{

            if(cursor.moveToFirst()) {

                for (int i=0; i < cursor.getCount(); i++){
                    int rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"));
                    String KataKor = cursor.getString(cursor.getColumnIndex("KataKor"));
                    String KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"));
                    String KataEng = cursor.getString(cursor.getColumnIndex("KataEng"));
                    KataItems items = new KataItems(rIndex, KataKor, KataIndo, KataEng);
                    list.add(items);
                    cursor.moveToNext();
                }
            }

        } catch (SQLiteException e){
        } catch (Exception e){
        } finally {

            if(cursor != null){
                cursor.close();
                cursor = null;
            }

            if(database != null){
                database.close();
            }
        }
        return list;
    }

    public ArrayList<KataItems> getSearchKata(String sSearch){

        if(databaseHelper == null){
            return null;
        }

        SQLiteDatabase database = databaseHelper.getReadableDatabase();
        if(database == null){
            return null;
        }

        ArrayList<KataItems> list = new ArrayList<>();
        Cursor cursor = null;

        try {

            //KataKor
            String query = "SELECT * from Katas where KataKor LIKE '" + sSearch + "%'";
            cursor = database.rawQuery(query, null);
            if(cursor.moveToFirst()) {

                for (int i=0; i < cursor.getCount(); i++){
                    int rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"));
                    String KataKor = cursor.getString(cursor.getColumnIndex("KataKor"));
                    String KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"));
                    String KataEng = cursor.getString(cursor.getColumnIndex("KataEng"));
                    //Log.e(TAG, "KataKor:" + KataKor);
                    KataItems items = new KataItems(rIndex, KataKor, KataIndo, KataEng);
                    list.add(items);
                    cursor.moveToNext();
                }
            }

            //KataIndo
            query = "SELECT * from Katas where KataIndo LIKE '" + sSearch + "%'";
            cursor = database.rawQuery(query, null);
            if(cursor.moveToFirst()) {

                for (int i=0; i < cursor.getCount(); i++){
                    int rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"));
                    String KataKor = cursor.getString(cursor.getColumnIndex("KataKor"));
                    String KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"));
                    String KataEng = cursor.getString(cursor.getColumnIndex("KataEng"));
                    //Log.e(TAG, "KataKor:" + KataKor);
                    KataItems items = new KataItems(rIndex, KataKor, KataIndo, KataEng);
                    list.add(items);
                    cursor.moveToNext();
                }
            }

            //KataEng
            query = "SELECT * from Katas where KataEng LIKE '" + sSearch + "%'";
            cursor = database.rawQuery(query, null);
            if(cursor.moveToFirst()) {

                for (int i=0; i < cursor.getCount(); i++){
                    int rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"));
                    String KataKor = cursor.getString(cursor.getColumnIndex("KataKor"));
                    String KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"));
                    String KataEng = cursor.getString(cursor.getColumnIndex("KataEng"));
                    //Log.e(TAG, "KataKor:" + KataKor);
                    KataItems items = new KataItems(rIndex, KataKor, KataIndo, KataEng);
                    list.add(items);
                    cursor.moveToNext();
                }
            }

        } catch (SQLiteException e){
        } catch (Exception e){
        } finally {

            if(cursor != null){
                cursor.close();
                cursor = null;
            }

            if(database != null){
                database.close();
            }
        }

        return list;
    }

    public int getHangulTotalcount(){

        int nTotalcount = 0;

        if(databaseHelper == null){
            return nTotalcount;
        }

        SQLiteDatabase database = databaseHelper.getReadableDatabase();
        if(database == null){
            return nTotalcount;
        }

        String query = "select count(*) from Hangul";
        Cursor cursor = database.rawQuery(query, null);

        try{

            if(cursor.moveToFirst()) {
                cursor.moveToFirst();
                nTotalcount =cursor.getInt(0);
            }

        } catch (SQLiteException e){
        } catch (Exception e){
        } finally {

            if(cursor != null){
                cursor.close();
                cursor = null;
            }

            if(database != null){
                database.close();
            }
        }
        return nTotalcount;
    }

    public HangulItems getHangul(int nIndex){

        HangulItems items = null;

        if(databaseHelper == null){
            return items;
        }

        SQLiteDatabase database = databaseHelper.getReadableDatabase();
        if(database == null){
            return items;
        }

        String query = "select * from Hangul where rIndex ="  + nIndex;
        Cursor cursor = database.rawQuery(query, null);

        try{

            if(cursor.moveToFirst()) {
                cursor.moveToFirst();
                int rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"));
                String KataFirst = cursor.getString(cursor.getColumnIndex("KataFirst"));
                String KataSecond = cursor.getString(cursor.getColumnIndex("KataSecond"));
                String KataEnd = cursor.getString(cursor.getColumnIndex("KataEnd"));
                items = new HangulItems(rIndex, KataFirst, KataSecond, KataEnd);
            }

        } catch (SQLiteException e){
        } catch (Exception e){
        } finally {

            if(cursor != null){
                cursor.close();
                cursor = null;
            }

            if(database != null){
                database.close();
            }
        }
        return items;
    }

}

File: bahasakorea(kotlin)/app/src/main/java/com/daeseong/bahasakorea/Database/KataManager.kt
package com.daeseong.bahasakorea.Database

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteException
import java.util.*

class KataManager {

    private val tag = KataManager::class.java.simpleName

    private val TABLE_KATAS = "Katas"

    companion object {

        private var databaseHelper: DatabaseHelper? = null

        private var instance: KataManager? = null
        fun getInstance(context: Context): KataManager {
            if (instance == null) {

                databaseHelper = DatabaseHelper(context)
                //databaseHelper = DatabaseHelper.getInstance(context)

                instance = KataManager()
            }
            return instance as KataManager
        }
    }

    fun setKata(items: KataItems) {

        if (databaseHelper == null) {
            return
        }

        val db  = databaseHelper!!.writableDatabase

        try {
            val values = ContentValues()
            values.put("rIndex", items.rIndex)
            values.put("KataKor", items.kataKor)
            values.put("KataIndo", items.kataIndo)
            values.put("KataEng", items.kataEng)
            db !!.insert(TABLE_KATAS, null, values)
        } catch (e: SQLiteException) {
        } catch (e: Exception) {
        } finally {
            db ?.close()
        }
    }

    fun getTotalcount(): Int {

        var nTotalcount = 0

        if (databaseHelper == null) {
            return nTotalcount
        }

        val database = databaseHelper!!.readableDatabase ?: return nTotalcount
        val query = "select count(*) from Katas"
        var cursor = database.rawQuery(query, null)

        try {
            if (cursor!!.moveToFirst()) {
                cursor.moveToFirst()
                nTotalcount = cursor.getInt(0)
            }
        } catch (e: SQLiteException) {
        } catch (e: java.lang.Exception) {
        } finally {
            if (cursor != null) {
                cursor.close()
                cursor = null
            }
            database?.close()
        }
        return nTotalcount
    }

    fun getKata(sSearch: String): KataItems? {

        var items: KataItems? = null

        if (databaseHelper == null) {
            return items
        }

        val database = databaseHelper!!.readableDatabase ?: return items
        val query = "select * from Katas where KataKor  =  \"$sSearch\""
        var cursor = database.rawQuery(query, null)

        try {
            if (cursor!!.moveToFirst()) {
                cursor.moveToFirst()
                val rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"))
                val KataKor = cursor.getString(cursor.getColumnIndex("KataKor"))
                val KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"))
                val KataEng = cursor.getString(cursor.getColumnIndex("KataEng"))
                items = KataItems(rIndex, KataKor, KataIndo, KataEng)
            }
        } catch (e: SQLiteException) {
        } catch (e: java.lang.Exception) {
        } finally {
            if (cursor != null) {
                cursor.close()
                cursor = null
            }
            database?.close()
        }
        return items
    }

    fun getKata(nIndex: Int): KataItems? {

        var items: KataItems? = null

        if (databaseHelper == null) {
            return items
        }

        val database = databaseHelper!!.readableDatabase ?: return items
        val query = "select * from Katas where rIndex =$nIndex"
        var cursor = database.rawQuery(query, null)

        try {
            if (cursor!!.moveToFirst()) {
                cursor.moveToFirst()
                val rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"))
                val KataKor = cursor.getString(cursor.getColumnIndex("KataKor"))
                val KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"))
                val KataEng = cursor.getString(cursor.getColumnIndex("KataEng"))
                items = KataItems(rIndex, KataKor, KataIndo, KataEng)
            }
        } catch (e: SQLiteException) {
        } catch (e: java.lang.Exception) {
        } finally {
            if (cursor != null) {
                cursor.close()
                cursor = null
            }
            database?.close()
        }
        return items
    }

    fun getKata(): ArrayList<KataItems>? {

        if (databaseHelper == null) {
            return null
        }

        val database = databaseHelper!!.readableDatabase ?: return null
        val list = ArrayList<KataItems>()
        val query = "select * from Katas "
        var cursor = database.rawQuery(query, null)

        try {
            if (cursor!!.moveToFirst()) {
                for (i in 0 until cursor.count) {
                    val rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"))
                    val KataKor = cursor.getString(cursor.getColumnIndex("KataKor"))
                    val KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"))
                    val KataEng = cursor.getString(cursor.getColumnIndex("KataEng"))
                    val items = KataItems(rIndex, KataKor, KataIndo, KataEng)
                    list.add(items)
                    cursor.moveToNext()
                }
            }
        } catch (e: SQLiteException) {
        } catch (e: java.lang.Exception) {
        } finally {
            if (cursor != null) {
                cursor.close()
                cursor = null
            }
            database?.close()
        }
        return list
    }

    fun getSearchKata(sSearch: String): ArrayList<KataItems>? {

        if (databaseHelper == null) {
            return null
        }

        val database = databaseHelper!!.readableDatabase ?: return null
        val list = ArrayList<KataItems>()
        var cursor: Cursor? = null

        try {

            //KataKor
            var query = "SELECT * from Katas where KataKor LIKE '$sSearch%'"
            cursor = database.rawQuery(query, null)
            if (cursor.moveToFirst()) {
                for (i in 0 until cursor.count) {
                    val rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"))
                    val KataKor = cursor.getString(cursor.getColumnIndex("KataKor"))
                    val KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"))
                    val KataEng = cursor.getString(cursor.getColumnIndex("KataEng"))
                    val items = KataItems(rIndex, KataKor, KataIndo, KataEng)
                    list.add(items)
                    cursor.moveToNext()
                }
            }

            //KataIndo
            query = "SELECT * from Katas where KataIndo LIKE '$sSearch%'"
            cursor = database.rawQuery(query, null)
            if (cursor.moveToFirst()) {
                for (i in 0 until cursor.count) {
                    val rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"))
                    val KataKor = cursor.getString(cursor.getColumnIndex("KataKor"))
                    val KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"))
                    val KataEng = cursor.getString(cursor.getColumnIndex("KataEng"))
                    val items = KataItems(rIndex, KataKor, KataIndo, KataEng)
                    list.add(items)
                    cursor.moveToNext()
                }
            }

            //KataEng
            query = "SELECT * from Katas where KataEng LIKE '$sSearch%'"
            cursor = database.rawQuery(query, null)
            if (cursor.moveToFirst()) {
                for (i in 0 until cursor.count) {
                    val rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"))
                    val KataKor = cursor.getString(cursor.getColumnIndex("KataKor"))
                    val KataIndo = cursor.getString(cursor.getColumnIndex("KataIndo"))
                    val KataEng = cursor.getString(cursor.getColumnIndex("KataEng"))
                    val items = KataItems(rIndex, KataKor, KataIndo, KataEng)
                    list.add(items)
                    cursor.moveToNext()
                }
            }
        } catch (e: SQLiteException) {
        } catch (e: java.lang.Exception) {
        } finally {
            if (cursor != null) {
                cursor.close()
                cursor = null
            }
            database?.close()
        }
        return list
    }

    fun getHangulTotalcount(): Int {

        var nTotalcount = 0

        if (databaseHelper == null) {
            return nTotalcount
        }

        val database = databaseHelper!!.readableDatabase ?: return nTotalcount
        val query = "select count(*) from Hangul"
        var cursor = database.rawQuery(query, null)

        try {
            if (cursor!!.moveToFirst()) {
                cursor.moveToFirst()
                nTotalcount = cursor.getInt(0)
            }
        } catch (e: SQLiteException) {
        } catch (e: java.lang.Exception) {
        } finally {
            if (cursor != null) {
                cursor.close()
                cursor = null
            }
            database?.close()
        }
        return nTotalcount
    }

    fun getHangul(nIndex: Int): HangulItems? {

        var items: HangulItems? = null

        if (databaseHelper == null) {
            return items
        }

        val database = databaseHelper!!.readableDatabase ?: return items
        val query = "select * from Hangul where rIndex =$nIndex"
        var cursor = database.rawQuery(query, null)

        try {
            if (cursor!!.moveToFirst()) {
                cursor.moveToFirst()
                val rIndex = cursor.getInt(cursor.getColumnIndex("rIndex"))
                val KataFirst = cursor.getString(cursor.getColumnIndex("KataFirst"))
                val KataSecond = cursor.getString(cursor.getColumnIndex("KataSecond"))
                val KataEnd = cursor.getString(cursor.getColumnIndex("KataEnd"))
                items = HangulItems(rIndex, KataFirst, KataSecond, KataEnd)
            }
        } catch (e: SQLiteException) {
        } catch (e: java.lang.Exception) {
        } finally {
            if (cursor != null) {
                cursor.close()
                cursor = null
            }
            database?.close()
        }
        return items
    }

}
